src/utils/transactions.ts

```diff
@@ -99,35 +99,27 @@ export const getDirection = (tx: Transaction, address: AddressHash): Transaction
   return amount && amountIsBigInt && amount < 0 ? 'out' : 'in'
 }
 
-export const calculateTotalTxOutput = (tx: Transaction) => {
-  const total = (tx.outputs ?? []).reduce((acc, o) => acc + BigInt(o?.attoAlphAmount ?? 0), BigInt(0))
-
-  // We must do best effort to determine the change addresses. Usually they are the last TXs.
-  const inputs = uniq((tx.inputs ?? []).map((i) => i.address))
-  let change = BigInt(0)
-
-  for (let index = (tx.outputs ?? []).length - 1; index > 0; index--) {
-    const output = (tx.outputs ?? [])[index]
-    const inputIndex = inputs.indexOf(output.address)
-    if (inputs[inputIndex]) {
-      change += BigInt(output.attoAlphAmount)
-      inputs.splice(inputIndex, 1)
-    }
-    if (inputs.length == 0) return total - change
-  }
+export const calculateAmountSent = (tx: Transaction | UnconfirmedTransaction): bigint => {
+  const outputs = tx.outputs ?? []
+  const inputAddresses = tx.inputs ? uniq(tx.inputs.map((input) => input.address)) : []
+
+  const total = outputs.reduce((acc, output) => acc + BigInt(output.attoAlphAmount), BigInt(0))
+  const change = outputs.reduce(
+    (acc, output) => (inputAddresses.includes(output.address) ? acc + BigInt(output.attoAlphAmount) : acc),
+    BigInt(0)
+  )
 
   return total - change
 }
 
-export const fromUnconfirmedTransactionToPendingTx = (
+export const convertUnconfirmedTxToPendingTx = (
   tx: UnconfirmedTransaction,
   belongingTo: AddressHash,
   network: NetworkName
 ): PendingTx => {
-  let amount = calculateTotalTxOutput(tx as unknown as Transaction)
-  const amountIsBigInt = typeof amount === 'bigint'
-  const type = amount && amountIsBigInt && amount < 0 ? 'out' : 'in'
-  amount = amount && (type === 'out' ? amount * BigInt(-1) : amount)
```

Comment:
As I mentioned in the other PR, `amount` will always be a `bigint` here. My changes reflect that.